dali/pipeline/operator/error_reporting.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATOR_ERROR_REPORTING_H_
+#define DALI_PIPELINE_OPERATOR_ERROR_REPORTING_H_
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "dali/core/api_helper.h"
+#include "dali/pipeline/data/types.h"
+#include "dali/pipeline/operator/op_spec.h"
+namespace dali {
+
+// TODO(klecki): Throw this one into a namespace?
+
+/**
+ * @brief Direct equivalent of Python's traceback.FrameSummary:
+ * https://docs.python.org/3/library/traceback.html#traceback.FrameSummary
+ *
+ */
+struct DaliFrameSummary {
```

Comment:
For consistency with the remaining 90% of the codebase
```suggestion
struct DALIFrameSummary {
```
or perhaps just drop it - it's in namespace `dali` anyway.

Reply:
I just took most of the naming from the Python struct that is linked in the docstring.
PythonStackFrame will be probably closest thing, even though I would be inclined to keep the name correspondence.